### DISCUSSION TITLE
LSP-like extension for links

### DISCUSSION THREAD

**Discussion by Aceticia** [nvim-neorg/neorg#512] [Ideas]
I'm writing my personal notes, journal, GTD, and wiki with neorg. With the telescope integration, things are seamless - I can search for content very effectively and link files. However, there are still some functionalities that are somewhat missing in maintaining a large scale wiki. Functions such as dead link detection, completion-based linking, previewing, links and backlinks, and renaming provided by markdown language servers such as [marksman](https://github.com/artempyanykh/marksman) would be a great compliment to the current, growing functionalities of neorg, for the people who take and maintain a lot of notes.

I don't know if any of these are planned content yet, since there does seem to be the zettelkasten module that just started. Is neorg always intended to be just for the organization side of things? Or does neorg have the ambition to achieve something like org-roam does?

**Comment by vhyrro** [comment]
An lsp-like extension would be fantastic and I'm fully down for someone to implement it (I'm lacking the time :p).
Neorg is supposed to be two things fused into one: a plugin to organize and a plugin to make editing .norg files easier. It's totally in the scope of Neorg to add integration or support for the features you're asking for, it's just a matter of when :)

**Reply by Aceticia** [reply]
That sounds awesome - if I get more time from my study I might try to help out a bit. If we will run the LSP feature continuously, I'm wondering whether the pure regex based approach implemented by telescope will be too slow in case of larger wiki collection? Will Neorg ever consider using some type of backend like sqlite to store the relations? Of course I'm not at that size yet, just wondering.

**Reply by vhyrro** [reply]
Although I can fairly confidently say that we will have to use some sort of backend eventually I'm undecided on what exactly I'd like to use. I don't know if i want the overhead of having to deal with sqlite databases (or have people run a DB in the background for that matter lol). Time will tell though!

**Reply by kluen** [reply]
Perhaps some inspiration can be taken from how [zk-nvim](https://github.com/mickael-menu/zk-nvim)/[zk](https://github.com/mickael-menu/zk) does this. They seem to be using sqlite as well.

Or a different thought: How about adding norg support (here's an [issue for org support](https://github.com/mickael-menu/zk/issues/143)) to zk and implementing an integration in nvim-neorg? I know this can probably not be a general solution, because it adds a dependency to that external project, but I can imagine an integration would be really great.
